File: frontend/src/app/core/services/mascota.service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { ApiService } from './api.service';
import { ApiResponse, PagedResponse } from '../models/api-response.model';
import { Mascota, MascotaRequest } from '../models/mascota.model';

@Injectable({ providedIn: 'root' })
export class MascotaService {
  constructor(private api: ApiService) {}

  getAll(params?: any): Observable<ApiResponse<PagedResponse<Mascota>>> {
    return this.api.getPaged('/mascotas', params?.page, params?.size, params);
  }

  getById(id: number): Observable<ApiResponse<Mascota>> {
    return this.api.get(`/mascotas/${id}`);
  }

  getByCliente(clienteId: number): Observable<ApiResponse<Mascota[]>> {
    return this.api.get(`/mascotas/cliente/${clienteId}`);
  }

  create(data: MascotaRequest): Observable<ApiResponse<Mascota>> {
    return this.api.post('/mascotas', data);
  }

  update(id: number, data: MascotaRequest): Observable<ApiResponse<Mascota>> {
    return this.api.put(`/mascotas/${id}`, data);
  }

  delete(id: number): Observable<ApiResponse<void>> {
    return this.api.delete(`/mascotas/${id}`);
  }
}

File: backend/src/main/java/bo/capital/tec/pet/mascota/service/MascotaService.java
package bo.capital.tec.pet.mascota.service;

import bo.capital.tec.pet.common.response.PagedResponse;
import bo.capital.tec.pet.mascota.dto.MascotaRequestDTO;
import bo.capital.tec.pet.mascota.dto.MascotaResponseDTO;
import bo.capital.tec.pet.mascota.dto.MascotaSummaryDTO;

import java.util.List;

public interface MascotaService {
    MascotaResponseDTO create(MascotaRequestDTO dto);
    MascotaResponseDTO getById(Long id);
    PagedResponse<MascotaSummaryDTO> getAll(String nombre, Long clienteId, Long especieId, int page, int size);
    List<MascotaSummaryDTO> getByClienteId(Long clienteId);
    List<MascotaSummaryDTO> getByUsuarioId(Long usuarioId);
    MascotaResponseDTO update(Long id, MascotaRequestDTO dto);
    void delete(Long id);
}
